lambdas/process-async-cri-credential/src/main/java/uk/gov/di/ipv/core/processasynccricredential/ProcessAsyncCriCredentialHandler.java

```diff
@@ -167,44 +168,47 @@ private void processErrorAsyncCriResponse(ErrorAsyncCriResponse errorAsyncCriRes
                         .with(
                                 LOG_ERROR_DESCRIPTION.getFieldName(),
                                 errorAsyncCriResponse.getErrorDescription())
-                        .with(LOG_ERROR_CODE.getFieldName(), errorAsyncCriResponse.getError())
-                        .with(
-                                LOG_CRI_ISSUER.getFieldName(),
-                                errorAsyncCriResponse.getCredentialIssuer()));
+                        .with(LOG_ERROR_CODE.getFieldName(), errorAsyncCriResponse.getError()));
 
-        sendIpvVcErrorAuditEvent(errorAsyncCriResponse);
+        if (criResponseItem.isPresent()
+                && Cri.F2F.equals(Cri.fromId(criResponseItem.get().getCredentialIssuer()))) {
+            sendIpvVcErrorAuditEvent(errorAsyncCriResponse);
+        }
     }
 
     private void processSuccessAsyncCriResponse(SuccessAsyncCriResponse successAsyncCriResponse)
             throws ParseException, CiPutException, AsyncVerifiableCredentialException,
                     CiPostMitigationsException, VerifiableCredentialException,
                     UnrecognisedVotException, CredentialParseException, EvcsServiceException,
                     HttpResponseExceptionWithErrorBody {
-        final CriResponseItem criResponseItem =
-                criResponseService.getCriResponseItem(
-                        successAsyncCriResponse.getUserId(),
-                        successAsyncCriResponse.getCredentialIssuer());
-        validateOAuthState(successAsyncCriResponse, criResponseItem);
+        var userId = successAsyncCriResponse.getUserId();
+        var state = successAsyncCriResponse.getOauthState();
+        Optional<CriResponseItem> criResponseItem = getCriResponseItem(userId, state);
+
+        if (criResponseItem.isEmpty()) {
+            LOGGER.error(LogHelper.buildLogMessage("No response item found"));
+            throw new AsyncVerifiableCredentialException(UNEXPECTED_ASYNC_VERIFIABLE_CREDENTIAL);
+        }
 
-        configService.setFeatureSet(criResponseItem.getFeatureSet());
+        var cri = Cri.fromId(criResponseItem.get().getCredentialIssuer());
```

Comment:
Where do we actually set the CRI id on the response item? I seem to remember we just assumed it's F2F, but this could become a problem if we need to retrieve and distinguish pending records in check existing identity 🤔

Reply:
It's set by the `call-dcmaw-async-cri` which produces a "pending" CRI response before you fill in details to get the VC which then appears on the queue. In this handler there is a section of code:

```java
criStoringService.recordCriResponse(
    request, DCMAW_ASYNC, oauthState, clientOAuthSessionItem, featureSets);
```

Which saves the cri response with `dcmawAsync` as the cri id. So it should be ok.

We did use to assume F2F when creating the `(success/error)AsyncCriResponse` object but you can see we have altogether removed that now as it is really just a dto for the queue response which doesn't include a cri id.